rclcpp/src/rclcpp/parameter_service.cpp

```diff
@@ -17,11 +17,78 @@
 #include <algorithm>
 #include <vector>
 
+#include "rcl/rcl.h"
+
+#include <string.h>
+
 using rclcpp::parameter_service::ParameterService;
 
 ParameterService::ParameterService(const rclcpp::node::Node::SharedPtr node)
 : node_(node)
 {
+  int argc;
+  char** argv;
+  argc = rcl_get_argv(&argv);
+  for (int i = 1; i < argc; ++i)
+  {
+    char * arg = strdup(argv[i]);
+    if (arg == NULL) {
+      break;
+    }
+    int len = strlen(arg);
+    // Minimum expression is _n:=v
```

Comment:
since [rcl_init()](http://docs.ros2.org/beta1/api/rcl/rcl_8h.html#a6abc5188c50b3a4a5fc41b357b49d6e5) requires the command line arguments to be passed to it, what do you think about parsing remapping rules in `rcl`? I think it would allow the same code to be reused across client libraries.